### PR TITLE
Validator Changes for Optimized Slasher

### PR DESCRIPTION
**Pull request by rauljordan**
This PR brings the validator changes from feature/slasher over into develop

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]

File: validator/client/attest_protect_test.go [hunk 1]
```diff
@@ -39,11 +38,23 @@ func Test_slashableAttestationCheck(t *testing.T) {
 			},
 		},
 	}
-	mockProtector := &mockSlasher.MockProtector{AllowAttestation: false}
-	validator.protector = mockProtector
+
+	m.slasherClient.EXPECT().IsSlashableAttestation(
+		gomock.Any(), // ctx
```
Comment (terencechain) [inline validator/client/attest_protect_test.go:43]: In most cases I would avoid using `gomock.Any()` and pass in the actual constructed objects

File: validator/client/propose.go [hunk 2]
```diff
@@ -270,9 +253,10 @@ func (v *validator) proposeBlockAltair(ctx context.Context, slot types.Slot, pub
 		}
 		return
 	}
-	if err := v.postBlockSignUpdate(ctx, pubKey, wsb, signingRoot); err != nil {
+
+	if err := v.slashableProposalCheck(ctx, pubKey, wsb, signingRoot); err != nil {
 		log.WithFields(
-			blockLogFields(pubKey, wb, sig),
+			blockLogFields(pubKey, wb, nil),
```
Comment (prestonvanloon) [inline validator/client/propose.go:259]: Good call. Leaking the sig into the logs could lead to someone sharing the logs and then someone else reconstructing the slashable offense.

File: validator/client/propose_protect_test.go [hunk 3]
```diff
@@ -91,64 +101,63 @@ func TestPreBlockSignLocalValidation(t *testing.T) {
...
-	require.ErrorContains(t, failedPostBlockSignErr, err, "Expected error when post signature update is detected as slashable")
-	mockProtector.AllowBlock = true
-	err = validator.postBlockSignUpdate(context.Background(), pubKey, wrapper.WrappedPhase0SignedBeaconBlock(emptyBlock), [32]byte{})
+
+	block := util.NewBeaconBlock()
+	block.Block.Slot = 10
+
+	m.slasherClient.EXPECT().IsSlashableBlock(
+		gomock.Any(), // ctx
+		gomock.Any(),
```
Comment (prestonvanloon) [inline validator/client/propose_protect_test.go:150]: Please use the actual block or something other than gomock.Any

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Please avoid use of gomock.Any(), except for context but even then sometimes you can use the correct ctx.

**Comment by prestonvanloon** [comment]
Ready for another look now. I discovered a panic condition in the previous release when the server would give a response where all of the slices were not the same length for whatever reason.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**